client/src/requests.rs

```diff
@@ -0,0 +1,1879 @@
+
+use std::str::FromStr;
+
+use jsonrpc::client::Request;//, Batch};
+use jsonrpc::client::{ConverterError, List, Param, Params};
+use serde::{Deserialize, Serialize};
+use serde_json::value::RawValue;
+
+use crate::bitcoin::hashes::hex::{HexIterator, FromHex};
+use crate::bitcoin::secp256k1::ecdsa::Signature;
+use crate::bitcoin::{
+    self, Address, Amount, Block, PrivateKey, PublicKey, ScriptBuf, Transaction,
+};
+use crate::bitcoin::block::Header as BlockHeader;
+use crate::bitcoin::psbt::PartiallySignedTransaction;
+type UncheckedAddress = Address<crate::bitcoin::address::NetworkUnchecked>;
+
+use crate::serialize::{
+    HexListSerializeWrapper, HexSerializeWrapper,
+    OutPointListObjectSerializeWrapper,
+    StringListSerializeWrapper, StringSerializeWrapper,
+};
+use crate::{json, AddressParam, BlockRef, BlockParam, Error, PsbtParam, SighashParam, TxParam};
+
+fn converter_hex<T>(raw: Box<RawValue>) -> Result<T, ConverterError>
+where
+    T: bitcoin::consensus::encode::Decodable,
+{
+    let hex: &str = serde_json::from_str(raw.get())?;
+    let mut bytes = HexIterator::new(hex)?;
+    Ok(T::consensus_decode(&mut bytes)?)
+}
+
+fn converter_raw_hex(raw: Box<RawValue>) -> Result<Vec<u8>, ConverterError> {
+    let hex: &str = serde_json::from_str(raw.get())?;
+    Ok(Vec::<u8>::from_hex(&hex)?)
+}
+
+pub(crate) fn converter_json<T>(raw: Box<RawValue>) -> Result<T, ConverterError>
+where 
+    T: serde::de::DeserializeOwned,
+{
+    Ok(serde_json::from_str(raw.get())?)
+}
+
+fn converter_btc(raw: Box<RawValue>) -> Result<Amount, ConverterError> {
+    let btc = serde_json::from_str::<f64>(raw.get())?;
+    Ok(Amount::from_btc(btc)?)
+}
+
+fn converter_psbt(raw: Box<RawValue>) -> Result<PartiallySignedTransaction, ConverterError> {
+    let b64 = serde_json::from_str::<&str>(raw.get())?;
+    Ok(PartiallySignedTransaction::from_str(b64)?)
+}
+
+/// Converter for calls that expect an empty result and a string indicates an error.
+fn converter_expect_null(raw: Box<RawValue>) -> Result<(), ConverterError> {
+    match serde_json::from_str::<serde_json::Value>(raw.get()) {
+        Ok(serde_json::Value::Null) => Ok(()),
+        Ok(res) => Err(Error::ReturnedError(res.to_string()).into()),
+        Err(err) => Err(err.into()),
+    }
+}
+
+// The following methods are shorthands to create [Param] objects.
+
+trait IntoJsonValue {
+    fn into_value(&self) -> serde_json::Value;
+}
+
+macro_rules! into_json_from {
+    ($t:ty) => {
+        impl IntoJsonValue for $t {
+            fn into_value(&self) -> serde_json::Value {
+                serde_json::Value::from(*self)
+            }
+        }
+    };
+}
+into_json_from!(bool);
+into_json_from!(usize);
+into_json_from!(u16);
+into_json_from!(u32);
+into_json_from!(u64);
+into_json_from!(i32);
+into_json_from!(i64);
+into_json_from!(f64);
+
+/// Convert a basic type into a value parameter.
+#[inline]
+fn v<T: IntoJsonValue>(v: T) -> Param<'static> {
```

Comment:
Yeah the main benefit was because I was typing them tens or hundreds of times. I can use rust-analyzer to change their names now though I guess. I agree they would be useful to have in rust-jsonrpc as constructors on the `Param` type. Like, API design is still more or less open for these two refactors, once I get some review/agreement on the design, we can try clean that up more. I do agree some more things can move into rust-jsonrpc.